backends/tfhe-cuda-backend/cuda/include/integer/integer.h

```diff
@@ -0,0 +1,421 @@
+#ifndef CUDA_INTEGER_H
+#define CUDA_INTEGER_H
+
+#include "../pbs/pbs_enums.h"
+#include <stdint.h>
+
+enum OUTPUT_CARRY { NONE = 0, GENERATED = 1, PROPAGATED = 2 };
+enum SHIFT_OR_ROTATE_TYPE {
```

Comment:
https://en.cppreference.com/w/cpp/language/enum

see potential examples at the very bottom with the enum class and : public some_integer_type which may help getting proper enums on the rust side (not necessarily in this PR, lots of moving parts already)


I would still go for enum class which enforces types safety on the C++ side vs enum which does not IIRC